tests/unit/test_resource_cleanup.py

```
@@ -0,0 +1,325 @@
+        file_name="stale2.txt",
+        is_uploaded=False,
+    )
+    stale2.created_at = datetime.now(UTC) - timedelta(hours=25)
+
+    db_session.add_all([stale1, stale2])
+    await db_session.commit()
+
+    # Mock storage service to fail on first resource, succeed on second
+    call_count = 0
+
+    def mock_stat_object(bucket: str, resource_id: str) -> None:
```

Ruff (0.15.0)

[warning] Unused function argument: `bucket`

(ARG001)